video/decode/vd_lavc.c

```diff
@@ -568,6 +572,19 @@ static void update_image_params(struct dec_video *vd, AVFrame *frame,
     vd_ffmpeg_ctx *ctx = vd->priv;
     struct MPOpts *opts = ctx->opts;
 
+#if HAVE_AVUTIL_MASTERING_METADATA
+    // Get the reference peak (for HDR) if available. This is cached into ctx
+    // when it's found, since it's not available on every frame (and seems to
+    // be only available for keyframes)
+    AVFrameSideData *sd = av_frame_get_side_data(frame,
+                          AV_FRAME_DATA_MASTERING_DISPLAY_METADATA);
+    if (sd) {
+        AVMasteringDisplayMetadata *mdm = (AVMasteringDisplayMetadata *)sd->data;
+        if (mdm->has_luminance)
+            ctx->cached_hdr_peak = av_q2d(mdm->max_luminance);
```

Comment:
That's what the `mdm->has_luminance` check is for, in theory. Although it's true that the values could be present but nonsensical.

Maybe we should have a sanity check on these values one way or the other, since files *are* floating around with broken metadata.